### ISSUE TITLE
rtsp to rtmp Delay up to about 30 seconds

### ISSUE THREAD

**Issue by hou718945431** [bytedeco/javacv#2020]
I use FFmpegFrameGrabber to process images frame by frame and push the stream to the rtmp server, but the rtmp playback delay is too high. What can I do to reduce this delay? I think I can accept it within 5 seconds

**Comment by saudet** [comment]
Did you make sure to use hardware acceleration everywhere?

**Comment by hou718945431** [comment]
My testing environment does not have a GPU. After using ffmpeg to block audio, the delay drops to about 3 seconds. Does JavaCV provide ffmpeg - an parameter configuration? how can i do

**Comment by hou718945431** [comment]
i use setTimestamp,  But it seems to have no effect

**Comment by saudet** [comment]
setTimestamp() doesn't affect audio frames, that's normal.

**Comment by hou718945431** [comment]
I want to use FFmpegFrameGrabber  by my own compiled ffmpeg and how can I do it

**Comment by saudet** [comment]
Build from source? ttps://github.com/bytedeco/javacpp-presets#manual-installation

**Comment by saudet** [comment]
Yes, most hardware acceleration is available in 1.5.8.

**Comment by hou718945431** [comment]
how can i do   ?
`grabber.setOption("hwaccel", "cuda"); 
`
and How do I know the decoder and encoder？h264_?

**Comment by saudet** [comment]
I don't believe hwaccel works, but setting video codecs with setVideoCodecName() does work:
https://trac.ffmpeg.org/wiki/HWAccelIntro

**Comment by hou718945431** [comment]
When I want to compile ffmpeg to support cuda, I need to tell him the installation path of cuda and how I need to inform it of the location in Javacv

**Comment by saudet** [comment]
You're talking about the --enable-cuda-nvcc build option? That's not currently part of the presets, but we can add it to the cppbuild.sh script.

**Comment by hou718945431** [comment]
> I don't believe hwaccel works, but setting video codecs with setVideoCodecName() does work: https://trac.ffmpeg.org/wiki/HWAccelIntro

i try to set  grabber.setVideoCodecName("h264_cuvid") and recorder.setVideoCodecName("h264_nvenc") get error,
`Error: [h264_cuvid @ 0xfffdbc1e06f0] Cannot load libnvcuvid.so.1
Error: [h264_cuvid @ 0xfffdbc1e06f0] Failed loading nvcuvid.
`
`
Error: [h264_nvenc @ 0xffff0c2c5ec0] Cannot load libnvidia-encode.so.1
`
how can i know what decoder and encoder it support

**Comment by saudet** [comment]
Those files come with the video drivers. You'll need to install the latest version of the video drivers for your GPU.
